tests/unit_tests/test_routes.py

```diff
@@ -221,6 +235,44 @@ def test_valid_kml_put(self):
         )
         self.compare_kml_contents(response, self.kml_dict["updated"])
 
+    def test_valid_gzipped_kml_put(self):
+        id_to_put = self.sample_kml['id']
+        with open('./tests/samples/updated-kml.xml.gz', 'rb') as updated_kml_file:
+            data = {
+                'admin_id':
+                    self.sample_kml['admin_id'],
+                'kml':
+                    (
+                        updated_kml_file,
+                        'updated-kml.xml.gz',
+                        'application/vnd.google-earth.kml+xml'
+                    )
+            }
+            response = self.app.put(
+                url_for('update_kml', kml_id=id_to_put),
+                data=data,
+                content_type="multipart/form-data",
+                headers=self.origin_headers["allowed"]
+            )
+        self.assertEqual(response.status_code, 200)
+        self.assertCors(response, ['DELETE', 'GET', 'HEAD', 'OPTIONS', 'PUT'])
+        self.assertEqual(response.content_type, "application/json")
+        for key in self.sample_kml:
+            # values for "updated" should and may differ, so ignore them in
+            # this assertion
+            if key != "updated":
+                self.assertEqual(self.sample_kml[key], response.json[key])
+
+        updated_item = self.dynamodb.Table(AWS_DB_TABLE_NAME).get_item(Key={
+            'kml_id': id_to_put
+        }).get('Item', None)
+        self.assertAlmostEqual(
+            datetime.datetime.fromisoformat(updated_item["updated"]).replace(tzinfo=None),
+            datetime.datetime.utcnow(),
+            delta=timedelta(seconds=0.3)
+        )
```

Comment:
Why reading the updated timestamp directly from dynamodb and not from the response ?

Reply:
Agreed, if we would fetch the `updated_item` from the dynamodb only for this assertion.
But as we need to fetch it anyways to do other assertions (e.g. on its content), we can as well use the timestamp data from the dynamodb for this assertion. Doesn't make a difference.
Otherwise, yes, you are right, we could have simply used the timestamp of the response.

Reply:
So I'd suggest to leave it this way. It is also used the same way in `test_valid_kml_put()` already.
OK with you?